### PR TITLE
Jest Tests

### PR DESCRIPTION
**Pull request by JOdiemo**
Write unit tests with React Testing Library.
Write unit tests with snapshots with Jest.

### CODE REVIEW

**Review by tresorsawasawa — COMMENTED** [review]
# STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @JOdiemo  ,:wave: :wave: 
While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 🛑

You have submitted a project that requires pair-programming but implemented by one contributor. Make both of you have some contributions on this PR before requesting another review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

---
Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

---
_Invalid Code Review Request **Does not count** into the code reviews limit._

File: src/__tests__/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  it('Test 9 button properly when 3 and x are already recorded ', () => {
+    const ACBtn = calculate({ total: '3', next: null, operation: 'x' }, '9');
+    expect(ACBtn).toEqual({ total: '3', next: '9', operation: 'x' });
+  });
+
+  it('Test 8 button properly when 4.33 are already recorded', () => {
+    const btn = calculate({ next: '4.33', total: null }, '8');
+    expect(btn).toEqual({ next: '4.338', total: null });
+  });
+});
```
Comment (danmainah) [inline src/__tests__/calculate.test.js]: - [ ] @JOdiemo and @H-Ramalan great work on adding tests for the calculate.js file, however you need to add all the test scenarios, check the if statements inside the method, each if statement is a scenario (the main if statement that its not inside another if statement).

File: src/__tests__/UserInt.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      act(() => {
+        render(<App />);
+      });
+      await userEvent.click(screen.getByRole('link', { name: /Home/i }));
+      expect(
+        screen.getByRole('heading', { name: /welcome to our page/i }),
+      ).toBeInTheDocument();
+    });
+  });
+});
```
Comment (danmainah) [inline src/__tests__/UserInt.test.js:32]: - [ ] @team Great work adding tests to some of your components, however, the project required testing of all components as shown below, kindly add tests for the remaining components.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @JOdiemo and @H-Ramalan , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Wrote unit tests for the files operate.js and calculate.js using Jest.
✔️  Used Jest snapshots to test the components.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by IngDavidTM — APPROVED** [review]
Hi Team 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The tests are passing ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**